pagopa/io-app
# feat: [IOBP-687,IOBP-885] Add recently used payment method from the payment flow

## Short description
This PR implements the ability to display the most recently used payment method during in-app payments, ensuring it matches the method used in a previous transaction.

## List of changes proposed in this pull request
- Updated the open API definitions;
- Added saga, action & reducer about the `recentUsedPaymentMethod`
- Added an additional list item at the top of the list that contains the most recently used payment method if available, otherwise it isn't showed
- Added a logic to remove the recently used payment method from the cluster membership

## How to test
- Checkout this PR: https://github.com/pagopa/io-dev-api-server/pull/415 from the io-dev-api-server and generate the API definitions;
- Start the io-dev-api-server;
- With the app in local env, start a payment flow from the "Payments" section screen tapping the CTA "Paga un avviso"
- Then tap on the CTA "Digita"
    - Inside of it, type any notice code -> go forward and type any fiscal code;
- After, you should be able to see a list of available payment method, complete the first payment choosing any payment method;
- When you complete the transaction, try to start another payment flow, you should be able to see the payment method previously you choose in the "Recently used" section;

## Preview

https://github.com/user-attachments/assets/62c30602-9581-42cd-b75c-dd3c0703aa22
